src/Razor/src/Microsoft.VisualStudio.RazorExtension/Microsoft.VisualStudio.RazorExtension.csproj

```diff
@@ -39,6 +39,12 @@
   <!-- Import the list of service hub services we proffer, to generate various files to include in the VSIX -->
   <Import Project="$(RepositoryEngineeringDir)targets\Services.props" />
 
+  <!-- Shared code -->
+  <ItemGroup>
+    <Compile Include="..\..\..\Compiler\shared\JsonReaderExtensions.cs" LinkBase="Shared" />
+    <Compile Include="..\..\..\Compiler\shared\TagHelperDescriptorJsonConverter.cs" LinkBase="Shared" />
```

Comment:
Consider using the either the [newer JSON.NET converters](https://github.com/dotnet/razor/tree/main/src/Shared/Microsoft.AspNetCore.Razor.Serialization.Json) or the [Message Pack formatters](https://github.com/dotnet/razor/tree/main/src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/MessagePack). I'd rather get rid of these super old JSON.NET converters that are currently only used by the compiler.

Reply:
Interesting, I didn't realize we had a different set of serializers. In this PR I specifically wanted to be able to use the results in the benchmarks, so will keep them the same for now, but I'll submit a separate PR to upgrade both simultaneously.

Reply:
I wrote faster implementations of the JSON.NET converters about a year ago that avoided a lot of the allocation in the older converters. (At that time, we already had different JSON.NET converters between compiler and tooling.)

So, if we're going to use JSON.NET converters in the tooling layer, I'd definitely prefer the newer ones.

Reply:
Sticking with Json makes sense to me, in case we ever need to tell users to send us the output for issue investigation purposes. It's not often we've asked them for project.razor.json files in the past, but its not zero either.